### PR TITLE
Tweak release issue template based on my first-run experience

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

I worked through a release issue using this template for the first time earlier this week. While doing so I noticed a few things that I (possibly subjectively) felt would make it easier to work through:

1. I think I was wrong about wrapping when I reviewed the original template. In practice I found that wrapped markdown comments are much easier to read, but wrapped markdown in the lists is actually pretty hard to follow when editing the issue. I found myself wanting the markdown list content to not be wrapped in the GitHub text box.
1. I found that having inline commentary with example links made it harder for my brain to immediately zero in on which link I needed to add next. I hope that keeping the examples but moving them into a precursor comment block will make it easier to open the issue and immediately identify which link needs to be populated next.
1. I found some of the 'front matter' around what version was being released and whether it was a minor or patch release to be a little repetitive, so I've tried to roll that all into the issue title.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

With my eyeballs looking at how this PR renders.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -7,97 +7,108 @@ labels: release
...
 
-I have:
-<!-- Please uncomment the following section only if cutting a minor release. These should be completed at the beginning of Code Freeze: https://crossplane.io/docs/v1.2/reference/release-cycle.html#code-freeze -->
 <!--
 - [ ] Created the [release branch][release-branch].
-- [ ] Created and merged an empty commit to the `master` branch
-  ([PR][empty-commit-pr]).
-- [ ] Run the [Tag workflow][rc-tag-workflow] on the `master` branch with the
-  next release candidate tag ([Tag][rc-tag]).
+- [ ] Created and merged an empty commit to the `master` branch ([PR][empty-commit-pr]).
```
Comment (negz) [inline .github/ISSUE_TEMPLATE/release.md:35]:
```suggestion
- [ ] Created and merged an empty commit to the `master` branch ([link][empty-commit-pr]).
```

I considered making this change too, so that it's super clear to readers which link in each bullet point will take them to the relevant artifact. Another option would be to ask assignee to simply paste links to the artifact of each step as comments. That would make life more difficult for the reader, but it would make life easier for the assignee. Currently you must continually edit, save, edit, save etc in the markdown of the issue description as you proceed through the release. I would have preferred to simply check off boxes in the issue description as I went and paste links to the relevant artifacts as distinct comments.

Comment (hasheddan) [reply .github/ISSUE_TEMPLATE/release.md:35]: @negz I typically do this by copying the markdown into my editor and making changes there, rather than trying to do it on the issue, which github does not give a great interface for. I would prefer not opting to use comments, but I am fine with changing the name of the link to `[link]`.

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -7,97 +7,108 @@ labels: release
...
+- [ ] Run the [Tag workflow][tag-workflow] on the relevant release branch with the proper release version ([Tag][tag]).
+- [ ] Run the [CI workflow][ci-workflow] on the release branch and verified that the tagged build version exists on the [releases.crossplane.io] `build` channel. ([build][release-build]).
+- [ ] Run the [Configurations workflow][configurations-workflow] on the release branch and verified  that version exists on [registry.upbound.io] for all getting started packages.
+      - [ ] [xp/getting-started-with-aws]
+      - [ ] [xp/getting-started-with-aws-with-vpc]
+      - [ ] [xp/getting-started-with-azure]
+      - [ ] [xp/getting-started-with-gcp]
+- [ ] Run the [Promote workflow][promote-workflow] with channel `stable` on the release branch and verified that the tagged build version exists on the [releases.crossplane.io] `stable` channel ([build][stable-build]).
+- [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes ([release][release]).
+- [ ] Updated the current release version in the [Crossplane docs website repo] ([PR][website-pr]).
```
Comment (hasheddan) [inline .github/ISSUE_TEMPLATE/release.md:48]: Would you mind putting this in a comment block stating it is only relevant for minor releases? I mistakenly did not do that in my earlier PR

File: .github/ISSUE_TEMPLATE/release.md [hunk 3]
```diff
@@ -7,97 +7,108 @@ labels: release
...
-    - [ ] [Slack][slack]
-    - [ ] [Twitter][twitter]
+- [ ] Updated all version information in the documentation on the relevant release branch ([PR][docs-update-pr]).
+- [ ] Run the [Tag workflow][tag-workflow] on the relevant release branch with the proper release version ([Tag][tag]).
+- [ ] Run the [CI workflow][ci-workflow] on the release branch and verified that the tagged build version exists on the [releases.crossplane.io] `build` channel. ([build][release-build]).
+- [ ] Run the [Configurations workflow][configurations-workflow] on the release branch and verified  that version exists on [registry.upbound.io] for all getting started packages.
+      - [ ] [xp/getting-started-with-aws]
+      - [ ] [xp/getting-started-with-aws-with-vpc]
+      - [ ] [xp/getting-started-with-azure]
+      - [ ] [xp/getting-started-with-gcp]
```
Comment (hasheddan) [inline .github/ISSUE_TEMPLATE/release.md:45]: These appear to be rendering weird for in the preview, does it look the same for you? I believe they may have an extra level of indentation.

File: .github/ISSUE_TEMPLATE/release.md [hunk 4]
```diff
@@ -7,97 +7,108 @@ labels: release
...
+- [ ] Run the [Promote workflow][promote-workflow] with channel `stable` on the release branch and verified that the tagged build version exists on the [releases.crossplane.io] `stable` channel ([build][stable-build]).
+- [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes ([release][release]).
+- [ ] Updated the current release version in the [Crossplane docs website repo] ([PR][website-pr]).
+- [ ] Updated the [releases table] in the `README.md` on `master` ([PR][release-table-pr]).
+- [ ] Ensured that users have been notified of the release on all communication channels:
+      - [ ] [Slack][slack]
+      - [ ] [Twitter][twitter]
```
Comment (hasheddan) [inline .github/ISSUE_TEMPLATE/release.md:52]: These also appear to render strangely for me.

**Review by hasheddan — COMMENTED** [review]
@negz thanks for being the first to try this out and for you feedback! Left a few comments below but I am on board with most of the changes :)

File: .github/ISSUE_TEMPLATE/release.md (hunk 3 above)
Comment (negz) [reply .github/ISSUE_TEMPLATE/release.md:45]: Ah yes - they are for me too. I added extra indentation to appease my markdown linter - I'll just revert that.

File: .github/ISSUE_TEMPLATE/release.md (hunk 2 above)
Comment (negz) [reply .github/ISSUE_TEMPLATE/release.md:48]: Will do.

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (negz) [reply .github/ISSUE_TEMPLATE/release.md:35]: Hrm, I'd personally prefer to lean into a flow that's more compatible with the native GitHub interface but I don't feel that strongly about it. Maybe we can have a few other folks work through the template and get feedback from them. For now I'll just make the `[link]` change.

Comment (hasheddan) [reply .github/ISSUE_TEMPLATE/release.md:35]: Are you saying that adding comments is more leaning into the GitHub interface flow? Since we can't have templates in comments I feel like this is a recipe for divergence from convention, potentially making the issues very difficult to parse when looking back on them.

Comment (negz) [reply .github/ISSUE_TEMPLATE/release.md:35]: I'm saying needing to leave GitHub and fill out the template in my IDE, then copy and paste it back into GitHub when I'm done does not seem like leaning into the GitHub UX very much. I can't imagine it's common or intuitive for folks to work that way with issues.

That said, this is not a big deal. I'm happy to defer until we have more folks with experience and opinions here.

**Review by hasheddan — COMMENTED** [review]
@negz i think you were going to push a few updates based on the comments on this PR, but otherwise LGTM!

**Comment by negz** [comment]
@hasheddan I just updated this with the review feedback.

**Review by hasheddan — APPROVED** [review]
